upcloud/problem.go

```diff
@@ -34,8 +41,37 @@ func (p *Problem) Error() string {
 	return sb.String()
 }
 
-// ProblemInvalidParam is a type describing extra information in the Problem type's InvalidParams field.
-type ProblemInvalidParam struct {
-	Name   string `json:"name"`
-	Reason string `json:"reason"`
+// MatchesProblemType is a helper method for comparing a `Problem.Type` field against a problem type constants defined in this package.
+// Please note that while you can compare the `Type` field directly to those constants, it is not recommended and might not always work reliably.
+// Examples:
+//
+//	res, err := svc.GetServerDetails(ctx, &req)
+//
+//	var problem *upcloud.Problem
+//	if errors.As(err, &problem) {
+//		// This is the recommended way
+//		if problem.MatchesProblemType(upcloud.ProblemTypeResourceAlreadyExists) {
+//			handleStuff(problem)
+//		}
+//
+//		// This might work, but is not guaranteed and not recommended
+//		if problem.Type == upcloud.ProblemTypeResourceAlreadyExists {
+//			handleStuff(problem)
+//		}
+//	}
+func (p *Problem) MatchesProblemType(problemType string) bool {
```

Comment:
The main reason why we need this is that the `Problem.Type` can be an URL that might change over time. For example now it's `https://developers.upcloud.com/1.3/errors#ERROR_SOME_ERROR_CODE`, but it might change in the future for operational reasons or something.

Technically we could add that whole URL string to constants and not have this method at all, but then if the actual URL in API response changes the SDK consumer code might stop working without any real warning. So I added this here, this is not really fully future-proof, but it seems a bit more secure to me.

Reply:
How about `func (p *Problem) ErrorCode() upcloud.ErrorCode`   ? 
I kind of liked more that naming that you had (if I remember correctly) previously `ErrorCodeResourceAlreadyExists` then we might have that as type `ErrorCode` e.g.:
```go
type ErrorCode string
const (
 ErrorCodeResourceAlreadyExists ErrorCode = "..."
 // ...
)
``` 
and usage would be something like:
```go
p.ErrorCode() == upcloud.ErrorCodeResourceAlreadyExists
```

Reply:
The reason why I changed that to `ProblemTypeXXX` is because... well problem doesn't have `Code` field or anything. But if we add a method like that then sure.
I guess this might also be a bit more go-proper to compare stuff like this

Reply:
Done